web3.js/src/transaction/versioned.ts

```diff
@@ -90,6 +92,27 @@ export class VersionedTransaction {
     return new VersionedTransaction(message, signatures);
   }
 
+  static fromLegacyTransaction(transaction: Transaction): VersionedTransaction {
+    const isUnsigned = transaction.signatures.every(signature => signature.signature == null);
+    if (isUnsigned && transaction.feePayer != null && transaction.recentBlockhash != null) {    
+      const message = new TransactionMessage({
+        instructions: transaction.instructions,
+        payerKey: transaction.feePayer,
+        recentBlockhash: transaction.recentBlockhash,
+      })
+      return new VersionedTransaction(message.compileToV0Message());
+    }
+
+    const message = transaction.compileMessage();        
```

Comment:
Can you say a little more about what's going on here? If the message is unsigned, then we recreate it as a `VersionedTransaction` with a version of `0`, but if it _is_ signed then we recreate it as a `VersionedTransaction` with a version of `legacy`?